### PR TITLE
Feat6 video page display

### PR DESCRIPTION
**Pull request by ShrutiLakhani**
1. Added Single Video Page
2. Add to History
3. Category Listing
4. Similar Video Recommendation

### CODE REVIEW

**Review by MaruthiPothuganti — APPROVED** [review]
Specified features work perfectly with good code. PR looks good to merged.

**Review by shashikantmaurya0007 — APPROVED** [review]
Good to be merged

File: src/App.js [hunk 1]
```diff
@@ -27,6 +28,7 @@ function App() {
         <Route path="/watchlater" element={<Watchlater />} />
         <Route path="/liked" element={<Liked />} />
         <Route path="/playlist" element={<Playlist />} />
+        <Route path="/video/:videoId" element={<VideoPage />} />
```
Comment (ashishkatiyar2404) [inline src/App.js:31]: Appreciation :-good to see u r using dynamic routes for VideoPage products/Items

File: src/Components/CategoryList/CategoryList.css [hunk 2]
```diff
@@ -0,0 +1,19 @@
+.category {
+  border: 1px solid var(--component-black-color);
+  padding: 2px 1rem;
+  border-radius: 2rem;
+  background-color: var(--component-container-color);
+  font-weight: 500;
+}
+.category-list {
+  /* margin-top: 2rem; */
```
Comment (ashishkatiyar2404) [inline src/Components/CategoryList/CategoryList.css:9]: you can remove comments

File: src/Components/CategoryList/CategoryList.jsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  const {
+    setFilteredList,
+    setFilteredVideos,
+    categories,
+    allVideos,
+    filterByCategory,
+    videoData,
+    videoDispatch,
+    filteredList,
+    setCurrentCategory,
```
Comment (ashishkatiyar2404) [inline src/Components/CategoryList/CategoryList.jsx:16]: you can remove which you are not using in this file.

File: src/Components/HistoryCard/HistoryCard.jsx [hunk 4]
```diff
@@ -1,7 +1,110 @@
-import React from "react";
+import React, { useState } from "react";
+import { useNavigate } from "react-router-dom";
+// import { removeItemFromLikedVideos } from "../../backend/controllers/LikeController";
```
Comment (ashishkatiyar2404) [inline src/Components/HistoryCard/HistoryCard.jsx:3]: here also you can remove comments

File: src/Components/LikedVideoCard/LikedVideoCard.jsx [hunk 5]
```diff
@@ -29,9 +29,18 @@ function LikedVideoCard({ video }) {
...
+    navigate(`/video/${_id}`);
+  };
+
   return (
     <div className="videocard-styling">
-      <img className="videocard-thumbnail" src={thumbnail} alt="" />
+      <img
+        className="videocard-thumbnail"
+        src={thumbnail}
+        alt=""
```
Comment (ashishkatiyar2404) [inline src/Components/LikedVideoCard/LikedVideoCard.jsx:41]: pls fill alt with proper value

File: src/Components/WatchlaterCard/WatchlaterCard.jsx [hunk 6]
```diff
@@ -27,9 +28,17 @@ function WatchlaterCard({ video }) {
...
+  const navigateToVideo = () => {
+    navigate(`/video/${_id}`);
+  };
   return (
     <div className="videocard-styling">
-      <img className="videocard-thumbnail" src={thumbnail} alt="" />
+      <img
+        className="videocard-thumbnail"
+        src={thumbnail}
+        alt=""
```
Comment (ashishkatiyar2404) [inline src/Components/WatchlaterCard/WatchlaterCard.jsx:39]: here also

File: src/Pages/VideoPage/VideoPage.jsx [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+  const suggestedList = suggestedVideos(allVideos, video);
+  return (
+    <div className="video-container">
+      <div className="video-pages">
+        <Sidebar />
+        <ReactPlayer
+          className="react-player-card"
+          controls={true}
+          width="1024px"
+          height="575px"
```
Comment (ashishkatiyar2404) [inline src/Pages/VideoPage/VideoPage.jsx:45]: you can avoid inline styling

File: src/context/video-context.js [hunk 8]
```diff
@@ -1,21 +1,29 @@
...
-  sortBy: "",
+  categoryList: [],
 };
 
 const VideoProvider = ({ children }) => {
+  const [videoData, videoDispatch] = useReducer(videoReducer, initialState);
   const [allVideos, setAllVideos] = useState([]);
+  const [currentCategory, setCurrentCategory] = useState("All");
+  const [historyVideo, setHistoryVideo] = useState([]);
+  // const [filteredList, setFilteredList] = useState(allVideos);
```
Comment (ashishkatiyar2404) [inline src/context/video-context.js:24]: pls remove comments

**Review by ashishkatiyar2404 — APPROVED** [review]
Your code functionality is good, I have suggested some changes pls look into them. 
Thn you can merge and work on other functionality of code

**Review by Varu98 — APPROVED** [review]
Clean Code. Nice!

**Pull request merged**